PropertyChanged.Fody.Analyzer.Tests/RoslynTestUtils.cs

```diff
@@ -0,0 +1,325 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/SourceGenerators/RoslynTestUtils.cs
+#pragma warning disable IDE0007 // Use implicit type
+#define ROSLYN4_0_OR_GREATER
+// ReSharper disable All
+
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SourceGenerators.Tests
+{
+    internal static class RoslynTestUtils
```

Comment:
Need to setup the project/compilation too, and I did not want to re-invent the wheel - if Microsoft is using this to test their generators, I thought it can't bee too bad. I just tried to keep it close to the original, so it has just a few stale methods, but it's only testing code.